cmd/controller/cmd/root.go

```diff
@@ -116,45 +112,39 @@ func logAndExit(err error) {
 	os.Exit(-1)
 }
 
-func getKubeConfig(_ context.Context, cfg *config2.Config) (*kubernetes.Clientset, *restclient.Config, error) {
-	var kubecfg *restclient.Config
-	var err error
-	if cfg.KubeConfigPath != "" {
-		kubeConfigPath := os.ExpandEnv(cfg.KubeConfigPath)
-		kubecfg, err = clientcmd.BuildConfigFromFlags(cfg.MasterURL, kubeConfigPath)
-		if err != nil {
-			return nil, nil, errors.Wrapf(err, "Error building kubeconfig")
-		}
-	} else {
-		kubecfg, err = restclient.InClusterConfig()
-		if err != nil {
-			return nil, nil, errors.Wrapf(err, "Cannot get InCluster kubeconfig")
-		}
-	}
-
-	kubecfg.QPS = cfg.KubeConfig.QPS
-	kubecfg.Burst = cfg.KubeConfig.Burst
-	kubecfg.Timeout = cfg.KubeConfig.Timeout.Duration
-
-	kubeClient, err := kubernetes.NewForConfig(kubecfg)
-	if err != nil {
-		return nil, nil, errors.Wrapf(err, "Error building kubernetes clientset")
-	}
-	return kubeClient, kubecfg, err
-}
-
 func sharedInformerOptions(cfg *config2.Config) []informers.SharedInformerOption {
+	labelSelector := controller.IgnoreCompletedWorkflowsLabelSelector()
+	addLabelSelector(labelSelector, transformers.ShardKeyLabel, v1.LabelSelectorOpIn, cfg.IncludeShardKeyLabel)
```

Comment:
how is this supposed to work?
So if we want to run for all - what do we do? Should these not be behind if conditions?

Reply:
the addLabelSelector function has a conditional inside checking if the shard label keys contains anything. change the name to addLabelSelectorIfExists?

Reply:
i changed the function name. we could otherwise define a labelSelector struct like we do in many of the tests to remove the addLabelSeletor function?

```
type labelSelector struct {
    label string
    operation v1.LabelSelectorOperator
    values []string
}

labelSelectors := []labelSelector{
    labelSelector{transformers.ShardKeyLabel, v1.LabelSelectorOpIn, cfg.IncludeShardKeyLabel},
    labelSelector{transformers.ShardKeyLabel, v1.LabelSelectorOpNot, cfg.ExcludeShardKeyLabel},
    ...
}

for _, labelSelector := range labelSelectors {
    if len(labelSelector.values) > 0 {
        // add label selector
    }
}
```